realtek: Whitespace and codestyle cleanup

oliv3r opened the pull request:
Fix some ugly whitepsaces and codestyle issues around the realtek sources.

Signed-off-by: Olliver Schinagl <oliver@schinagl.nl>

@svanheule this is now only targeting v5.15 (label?)

oliv3r committed:
realtek: Whitespace and codestyle cleanup

Fix some ugly whitepsaces and codestyle issues around the realtek sources.

While this is by no means perfect, it catches what it caught.

Signed-off-by: Olliver Schinagl <oliver@schinagl.nl>

Ansuel commented:
@oliv3r LGTM just i would split this to 2 or 4 commits? One specific to driver and other for the mk?

Also I wonder if this can be improved even more by making the realtek driver a patch and run it under the check patch script on linux kernel?

But they are just suggestions.

oliv3r commented:
> @oliv3r LGTM just i would split this to 2 or 4 commits? One specific to driver and other for the mk? (Also i would drop the cosmetic tag since it can be confusing)

Yeah I can split it up; I think 2 makes sense. 1 for the 'kernely bits' e.g. everything under files; one for the build stuff. Good suggestion (though It was nice having it as a single one while working on it ;)

As for the cosmetic tag, I kind of hoped it would trigger the label to be added. Regardless, I'll remove it if it pleases :)

Ansuel commented:
> As for the cosmetic tag, I kind of hoped it would trigger the label to be added. Regardless, I'll remove it if it pleases :)

label are added based on the change. Yes it would be good if it can be dropped with better title

oliv3r committed:
realtek: Cleanup Makefiles

Our current Makefiles a little bit messy and can be improved somewhat,
both in whitespace and in style.

Signed-off-by: Olliver Schinagl <oliver@schinagl.nl>

oliv3r committed:
realtek: Replace C++ style comments

The only exception to C++ style comments are SPDX license identifier
markers at the start of C files (even headers have C style markers).

Signed-off-by: Olliver Schinagl <oliver@schinagl.nl>

oliv3r committed:
realtek: Reduce variable scopes

Linus prefers to have loop initializers nice and tightly scoped. In
OpenWRT this has been possible since 41a1a652fbd4 ("kernel: backport
gnu11 upgrade").

This patch cleans up variable scope while trying to do the above for
'simple for loops'.

This cleans up and simplifies some functions and code, and pulls in
variables to a smaller scope.

Signed-off-by: Olliver Schinagl <oliver@schinagl.nl>

Ansuel commented on target/linux/realtek/files-5.15/drivers/net/dsa/rtl83xx/qos.c:
```diff
@@ -539,15 +544,17 @@ void rtl839x_config_qos(void)
 	sw_w32((0x1 << 2) | (0x1 << 4), RTL839X_RMK_DEI_CTRL);
 
-	/* Set Congestion avoidance drop probability to 0 for drop precedences 0-2 (bits 24-31)
+	/*
```
comments on netdev related stuff asks for comments starting from /* so this comment format is ok.

oliv3r replied:
Wait, you are saying that we have linus's coding style, and netdev coding style? I'm pretty sure Linus has in the past raged at people for using the wrong stye (I'd find Linus's mail if you really want it :p).

Ansuel replied:
Yep kernel checkpatch script apply different rules for comments related to netdev code. Mainly it's in the first line with multi line comments. 

Happy to read the email to have some fun with Linus getting angry ahahahah

oliv3r replied:
That is so ridiculous. I actually found the same statement in the coding style guide. Like 'wtf'. Why can you not just be consistent.

As for finding his post, needle, haystack :( It was something along the lines of 'This is how you write a comment'. Anyway, I'll remove all the 'multi-line comments starting with a near blank line' altogether from 'files/net' which is nearly everything then.

Ansuel commented on target/linux/realtek/files-5.15/drivers/net/dsa/rtl83xx/rtl838x.c:
```diff
@@ -316,7 +316,7 @@ static void rtl838x_fill_l2_entry(u32 r[], struct rtl838x_l2_entry *e)
-			
+
 			/* A valid entry has one of mutli-cast, aging, sa/da-blocking,
 			 * next-hop or static entry bit set */
```
Ideally this should be 
```
/* TEXT
 * TEXT
 */
```

Ansuel commented on target/linux/realtek/files-5.15/drivers/net/dsa/rtl83xx/rtl838x.c:
```diff
@@ -453,8 +450,7 @@ static u64 rtl838x_read_cam(int idx, struct rtl838x_l2_entry *e)
 	pr_debug("Found in CAM: R1 %x R2 %x R3 %x\n", r[0], r[1], r[2]);
 
 	// Return MAC with concatenated VID ac concatenated ID
```
and this kind should be /* TEXT */

Ansuel commented on target/linux/realtek/files-5.15/drivers/net/dsa/rtl83xx/rtl838x.h:
```diff
@@ -398,7 +398,7 @@
 #define RTL839X_SPCL_TRAP_SWITCH_IPV4_ADDR_CTRL	(0x106C)
 #define RTL839X_SPCL_TRAP_CRC_CTRL		(0x1070)
 /* special port action controls */
-/* 
```
aaand 
```
/*
 *
 *
 ...
```

oliv3r commented:
@Ansuel sharp eye :) I probably missed some, as later in my branch, I move some stuff around, which get 'automatically fixed'. Also I may have overlooked things naturally. It's a big patch after all (sadly).

svanheule commented:
@ansuel, thanks for the extra pair of eyes! Should I add your Reviewed-by: to the patches?

oliv3r commented:
@svanheule @Ansuel I did the 'netdev' style comments on all 'net' stuff. So that should address all your comments!

svanheule commented:
@oliv3r I noticed you have a number of `int i= 0` instances in "realtek: Reduce variable scopes", as well as `u16 data =0 , data_m = 0;`. If you like, I can fix these to have spaces around the "=" when merging. Wouldn't make much sense to me to do these clean-up patches, and then keep stuff like that.

oliv3r commented:
> @oliv3r I noticed you have a number of `int i= 0` instances in "realtek: Reduce variable scopes", as well as `u16 data =0 , data_m = 0;`. If you like, I can fix these to have spaces around the "=" when merging. Wouldn't make much sense to me to do these clean-up patches, and then keep stuff like that.

Let me grep for that and fix it right away.

oliv3r commented:
@svanheule should be good now (unless my grep-foo has been disappointing again :p

Ansuel commented:
Thanks for the multiple fixup btw!

Il Mar 27 Dic 2022, 15:59 Olliver Schinagl ***@***.***> ha
scritto:

> @svanheule <https://github.com/svanheule> should be good now (unless my
> grep-foo has been disappointing again :p
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/openwrt/openwrt/pull/11572#issuecomment-1365963847>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AE2ZMQV7BUSPJKWDPD6YZXDWPL75TANCNFSM6AAAAAATEOGK5Q>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

oliv3r commented:
> Thanks for the multiple fixup btw!

Heh, the whole point of this MR was to clean shit up; so lets do just that :)

It started out as 'oh lets just fix this' which added 'oh and that too' which turned out to be this massive undertaking.

I hope we can merge this now, can always do follow ups later (unless I missed something obvious, I don't mind doing yet more fixup, if something gets spotted now).

svanheule commented:
Rebased on some other changes and merged with:

758c88b96963 realtek: Whitespace and codestyle cleanup
0a931767cf2a realtek: Replace C++ style comments
94d8b4852b9f realtek: Cleanup Makefiles
0a83889e89b7 realtek: Reduce variable scopes

oliv3r commented:
> Rebased on some other changes and merged with:

Curious what you rebased ontop of; as I rebase very often :)

svanheule commented:
Jan's four patches that I merged together with yours, so you never had the chance to rebase on top of those :)

Was just some updated comments, where I had to re-apply the styling.
